core/util/sanctionsearch.py

```diff
@@ -0,0 +1,174 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import json
+from bs4 import BeautifulSoup as bs
+
+class main:
+
+		def __init__(self, name, id=None, typ='individual', limit=15):
+			""" Sanctionsearch.ofac.treas.gov search engine
+					name     : name to search
+			"""
+			self.framework = main.framework
+			self._name = name
+			self._type = typ
+			self._listid = 'ALL'
+			self._max = limit
+			self._id = id
+			self._json = ''
+			self.sanctionsearch = 'https://sanctionssearch.ofac.treas.gov/'
+			self._rows = []
+
+			if self._name is not None:
+				self._data = []
+			else:
+				self._data = {}
+
+
+		def name_crawl(self):
+			self._name = self.framework.urlib(self._name).quote
+			self.framework.verbose('Searching sanctionsearch...')
+			headers = {
+				'Content-Type': 'application/x-www-form-urlencoded',
+				}
+
+			try:
+				# First request to get viewstate from input tag
+				req = self.framework.request(url=self.sanctionsearch)
+				soup = bs(req.text,'html.parser')
+				viewstate = soup.find('input',
```

Comment:
after except block. just request can be in try

Reply:
There are two requests.

Reply:
So two try except blocks?